fe/fe-core/src/main/java/com/starrocks/sql/analyzer/BackupStmtAnalyzer.java

```diff
@@ -0,0 +1,189 @@
+            if (repo == null) {
+                ErrorReport.reportSemanticException(ErrorCode.ERR_COMMON_ERROR, "Repository does not exist");
+            }
+
+            Map<String, TableRef> tblPartsMap = Maps.newTreeMap(String.CASE_INSENSITIVE_ORDER);
+            List<TableRef> tableRefs = statement.getTableRefs();
+            for (TableRef tableRef : tableRefs) {
+                TableName tableName = tableRef.getName();
+                PartitionNames partitionNames = tableRef.getPartitionNames();
+                Table tbl = db.getTable(tableName.getTbl());
+                if (tbl == null) {
+                    ErrorReport.reportSemanticException(ErrorCode.ERR_WRONG_TABLE_NAME, tableName.getTbl());
+                }
+
+                if (!Strings.isNullOrEmpty(tableName.getDb())) {
+                    throw new SemanticException("Cannot specify database name on backup objects: "
+                            + tableName.getTbl() + ". Sepcify database name before label");
+                }
+
+                tableName.setCatalog(catalog);
+                tableName.setDb(dbName);
+                if (partitionNames != null) {
+                    if (tbl.getType() != Table.TableType.OLAP) {
+                        ErrorReport.reportSemanticException(ErrorCode.ERR_WRONG_TABLE_NAME, tableName.getTbl());
+                    }
+
+                    OlapTable olapTbl = (OlapTable) tbl;
+                    for (String partName : tableRef.getPartitionNames().getPartitionNames()) {
+                        Partition partition = olapTbl.getPartition(partName);
+                        if (partition == null) {
+                            throw new SemanticException(
+                                    "partition[" + partName + "] does not exist in table" + tableName.getTbl());
+                        }
+                    }
+                }
+
+                if (!tblPartsMap.containsKey(tableName.getTbl())) {
+                    tblPartsMap.put(tableName.getTbl(), tableRef);
+                } else {
+                    throw new SemanticException("Duplicated restore table: " + tableName.getTbl());
+                }
+
+                try {
+                    CatalogUtils.checkIsLakeTable(dbName, tableName.getTbl());
+                } catch (AnalysisException e) {
+                    throw new SemanticException(e.getMessage());
+                }
+
+                if (tableRef.hasExplicitAlias()) {
+                    throw new SemanticException("Can not set alias for table in Backup Stmt: " + tableRef);
+                }
+            }
+
+            tableRefs.clear();
+            for (TableRef tableRef : tblPartsMap.values()) {
+                statement.getTableRefs().add(tableRef);
+            }
+
+            Map<String, String> properties = statement.getProperties();
+            long timeoutMs = 0L;
+            if (properties.containsKey("timeout")) {
```

Comment:
Why not just iterate the properties and check the key?

Reply:
Is there any difference between rarely using iterators for this property and not using iterators at all

Reply:
If using a for loop to iterate the properties, you don't need to copy the properties to do some trick.

And it would be extensible for more properties.